core/src/main/scala/org/http4s/blaze/pipeline/Stages.scala

```diff
@@ -314,17 +322,21 @@ trait MidStage[I, O] extends Tail[I] with Head[O] {
 
   final def replaceNext(stage: LeafBuilder[O]): Tail[O] = _nextStage.replaceInline(stage)
 
-  final def removeStage(implicit ev: MidStage[I,O] =:= MidStage[I, I]): this.type = {
+  final def removeStage(implicit ev: MidStage[I,O] =:= MidStage[I, I]): Unit = {
     stageShutdown()
 
-    if (_prevStage == null || _nextStage == null) return this
+    if (_prevStage != null && _nextStage != null) {
+      val me: MidStage[I, I] = ev(this)
+      _prevStage._nextStage = me._nextStage
+      me._nextStage._prevStage = me._prevStage
 
-    val me: MidStage[I, I] = ev(this)
-    _prevStage._nextStage = me._nextStage
-    me._nextStage._prevStage = me._prevStage
-
-    _nextStage = null
-    _prevStage = null
-    this
+      _nextStage = null
+      _prevStage = null
+    }
+    else {
+      val e = new Exception("Cannot remove a disconnected stage")
```

Comment:
I'm not sure this needs to be an exception: the end result is the stage isn't there.  I'd long a stern warning.